server/src/packages/landuse-form/landuse-actions/landuse-actions.controller.ts

```diff
@@ -28,15 +28,24 @@ export class LanduseActionsController {
 
   @Patch('/:id')
   async update(@Body() body, @Param('id') id) {
-    const allowedAttrs = pick(body, LANDUSE_ACTION_ATTRS);
+    let allowedAttrs = pick(body, LANDUSE_ACTION_ATTRS);
     const dateOfPreviousApproval = this.pickDate(body.dcp_dateofpreviousapproval);
     const lapseDateOfPreviousApproval = this.pickDate(body.dcp_lapsedateofpreviousapproval);
     const recordationDate = this.pickDate(body.dcp_recordationdate);
 
+    // unlike other attributes, chosen_zoning_resolution_id
+    // is not always included in the body. So we make an
+    // explicit check for "null".
+    // TODO: Investigate why
+    if (body.chosen_zoning_resolution_id) {
```

Comment:
The comment here could do better at explaining the problem. I like to start with a premise and then get into the problem:

> "chosen_zoning_resolution" is a related resource, although it may not always be included in the request body:
> 
> "If a request does not include all of the relationships for a resource, the server MUST interpret the missing relationships as if they were included with their current values. It MUST NOT interpret them as null or empty values." See https://jsonapi.org/format/#crud-updating-resource-relationships.
>
> Therefore, we explicitly check for a `null` value (which would imply a key for `chosen_zoning_resolution_id`), to avoid assuming that falsiness indicates a disassociation.

Most of the time, I try to name methods and variables in a self-explanatory way. If that fails, we should prefer long comments.

Reply:
Thank you I can add these details about JSON API. The weird part here for me is why Ember Data decides to include most unmodified properties, but not this one specifically.

Reply:
@godfreyyeung well, recall there's a deserialization step involved. Ember data doesn't actually provide the relationship ID in that way (directly on the body). See JSON API deserializer class.